homeassistant/components/homeassistant/scene.py

```diff
@@ -139,7 +144,24 @@ def _convert_states(states):
 
     async def create_service(call):
         """Create a scene."""
-        scene_config = SCENECONFIG(call.data[CONF_SCENE_ID], call.data[CONF_ENTITIES])
+        snapshot = call.data[CONF_SNAPSHOT]
+        entities = call.data[CONF_ENTITIES]
+
+        for entity_id in snapshot:
+            state = hass.states.get(entity_id)
+            if state is None:
+                _LOGGER.warning(
+                    "Entity %s does not exist and therefore cannot be snapshooted",
+                    entity_id,
+                )
+                continue
+            entities[entity_id] = State(entity_id, state.state, state.attributes)
```

Comment:
Will it be confusing to the user that snapshot entity_ids may override entity_ids from `CONF_ENTITIES`?

Reply:
let's make it a validation error that they can't overlap

Reply:
Is it possible to use voluptuous for this?

Reply:
Yes. Write a custom validator function that checks the entity_ids in the two containers. See eg:
https://github.com/home-assistant/home-assistant/blob/6d77c15f2805d280c605ed9a669a9f5ecda640b2/homeassistant/components/mysensors/__init__.py#L40-L52

https://github.com/home-assistant/home-assistant/blob/6d77c15f2805d280c605ed9a669a9f5ecda640b2/homeassistant/components/mysensors/__init__.py#L100-L102

Reply:
Thanks :+1: